### ISSUE TITLE
Support prisma2@preview23 improvements

### ISSUE THREAD

**Issue by Weakky** [graphql-nexus/nexus-plugin-prisma#622]
### Description

While we're about to drop support for `preview23` on the surface, we haven't tested any of the major improvements yet.

This includes:

- Models without IDs
- Composite primary keys
- Composite unique indexes / constraints

**Comment by keita-makino** [comment]
I have encountered the issue where there is id field associated with another type.
Even though I didn't change the schema.ts at all it started not working after I introduce the following prisma model.

```prisma
model User {
  email    String  @default("") @unique
  id       Int     @default(autoincrement()) @id
  name     String  @default("")
  password String  @default("")
  setting  Setting
}

model Setting {
  id       User  @id
  something Float @default(0.00025)
  another Float @default(5)
}
```

```node
Error: Could not find field 'Query.undefined' on type Query
    at OutputType.getField (C:\Users\[path]\server\node_modules\nexus-prisma\src\dmmf\DmmfDocument.ts:130:13)
    at utils_1.flatMap.map.mappedField (C:\Users\[path]\server\node_modules\nexus-prisma\src\mapping.ts:56:41)
    at Array.map (<anonymous>)
    at Object.exports.getCrudMappedFields (C:\Users\[path]\server\node_modules\nexus-prisma\src\mapping.ts:52:64)
    at renderNexusPrismaInputs (C:\Users\[path]\server\node_modules\nexus-prisma\src\typegen.ts:117:25)
    at render (C:\Users\[path]\server\node_modules\nexus-prisma\src\typegen.ts:46:3)
    at doGenerate (C:\Users\[path]\server\node_modules\nexus-prisma\src\typegen.ts:27:25)
    at Object.generateSync (C:\Users\[path]\server\node_modules\nexus-prisma\src\typegen.ts:13:3)
    at new SchemaBuilder (C:\Users\[path]\server\node_modules\nexus-prisma\src\builder.ts:240:15)
    at Object.build (C:\Users\[path]\server\node_modules\nexus-prisma\src\builder.ts:138:19)
[ERROR] 20:34:28 Error: Could not find field 'Query.undefined' on type Query
```

**Comment by Weakky** [comment]
This feature was implemented in #686.

We've just released nexus-prisma@0.15.0-next.1.
You can try the feature by installing nexus-prisma@next right now. Feel free to report any bug related to it in here. I'll re-open the issue if needed 🙏
